### PR TITLE
Feature/private routing

### LINKED ISSUE #24: There is dataConverter.ts in @middleware/helpers. Consider placing this helper function there.
There is dataConverter.ts in @middleware/helpers. Consider placing this helper function there.

_Originally posted by @marcioszka in https://github.com/AlexanderKrysiuk/LinkUp-Frontend/pull/21#discussion_r1327959528_

### LINKED ISSUE #25: There is authHandler.ts in @middleware for all methods related to user authorization/athentication. You may move the function there :)
There is authHandler.ts in @middleware for all methods related to user authorization/athentication. You may move the function there :)

_Originally posted by @marcioszka in https://github.com/AlexanderKrysiuk/LinkUp-Frontend/pull/21#discussion_r1327959741_

### PR DESCRIPTION
**Pull request by Zachiel**
Implemented Secured Routing for Enhanced User Authentication

- Introduced secured routes to ensure proper user authentication.
- Enhanced user experience by securing access to sensitive pages.
- Addressed minor issues related to content rendering.

This pull request introduces secured routing, providing better control over user access to certain pages. It improves user authentication and enhances the overall user experience.

### CODE REVIEW

File: src/router/SecuredRoute.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	user: RoleEnum,
+	requiredRole: RoleEnum,
+): boolean => {
+	return user >= requiredRole;
+};
+
+const convertRole = (role: string): RoleEnum => {
+	const convertedRole = RoleEnum[role as keyof typeof RoleEnum];
+	return convertedRole;
+};
```
Comment (marcioszka) [inline src/router/SecuredRoute.tsx]: There is dataConverter.ts in @middleware/helpers. Consider placing this helper function there.

File: src/router/SecuredRoute.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+	return userRole !== null && userRole >= authRole ? children : null;
+};
+
+const checkAuthorization = (
+	user: RoleEnum,
+	requiredRole: RoleEnum,
+): boolean => {
+	return user >= requiredRole;
+};
```
Comment (marcioszka) [inline src/router/SecuredRoute.tsx]: There is authHandler.ts in @middleware for all methods related to user authorization/athentication. You may move the function there :)

**Review by marcioszka — APPROVED** [review]
All is good, just consider placing some methods in middleware.

**Comment by Zachiel** [comment]
Resolved issues #24 and #25

**Pull request merged**